### PR TITLE
Few trivias

### LINKED ISSUE #157: Sets should not appear in LaymanError data
Version: https://github.com/jirik/layman/commit/af4a0d5e723ca4749bf8b9da696dd27a76ae7f26

```
E               Traceback (most recent call last):
E                 File "/code/src/layman/common/prime_db_schema/publications.py", line 126, in check_publication_info
E                   info['access_rights'].get('write_old'),
E                 File "/code/src/layman/common/prime_db_schema/publications.py", line 105, in check_rights_axioms
E                   only_valid_names(can_read)
E                 File "/code/src/layman/common/prime_db_schema/publications.py", line 73, in only_valid_names
E                   raise LaymanError(43, {f'Not existing username. Username={username}'})
E               layman.http.LaymanError: LaymanError code=43 message=Wrong access rights. data={'Not existing username. Username=test_gs_rules_user'} private_info=None
E               
E               During handling of the above exception, another exception occurred:
E               
E               Traceback (most recent call last):
E                 File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1950, in full_dispatch_request
E                   rv = self.dispatch_request()
E                 File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1936, in dispatch_request
E                   return self.view_functions[rule.endpoint](**req.view_args)
E                 File "/code/src/layman/layer/rest_layers.py", line 165, in post
E                   raise e
E                 File "/code/src/layman/layer/rest_layers.py", line 157, in post
E                   'layman.layer.filesystem.input_chunk' if use_chunk_upload else 'layman.layer.filesystem.input_file'
E                 File "/code/src/layman/layer/util.py", line 162, in post_layer
E                   call_modules_fn(sources, 'post_layer', [username, layername], kwargs=task_options)
E                 File "/code/src/layman/util.py", line 181, in call_modules_fn
E                   k: v for k, v in kwargs.items()
E                 File "/code/src/layman/layer/prime_db_schema/table.py", line 57, in post_layer
E                   pubs_util.insert_publication(username, db_info)
E                 File "/code/src/layman/common/prime_db_schema/publications.py", line 151, in insert_publication
E                   check_publication_info(workspace_name, info)
E                 File "/code/src/layman/common/prime_db_schema/publications.py", line 136, in check_publication_info
E                   'message': exc_info.data,
E               layman.http.LaymanError: LaymanError code=43 message=Wrong access rights. data={'workspace_name': 'test_gs_rules_user', 'publication_name': 'test_gs_rules_layer', 'access_rights': {'read': ['test_gs_rules_user', 'EVERYONE'], 'write': ['test_gs_rules_user', 'EVERYONE']}, 'actor_name': None, 'owner': None, 'message': {'Not existing username. Username=test_gs_rules_user'}} private_info=None
E               
E               During handling of the above exception, another exception occurred:
E               
E               Traceback (most recent call last):
E                 File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 2464, in __call__
E                   return self.wsgi_app(environ, start_response)
E                 File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 2450, in wsgi_app
E                   response = self.handle_exception(e)
E                 File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1867, in handle_exception
E                   reraise(exc_type, exc_value, tb)
E                 File "/usr/local/lib/python3.6/dist-packages/flask/_compat.py", line 39, in reraise
E                   raise value
E                 File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 2447, in wsgi_app
E                   response = self.full_dispatch_request()
E                 File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1952, in full_dispatch_request
E                   rv = self.handle_user_exception(e)
E                 File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1822, in handle_user_exception
E                   return handler(e)
E                 File "/code/src/layman/__init__.py", line 104, in handle_invalid_usage
E                   response = jsonify(error.to_dict())
E                 File "/usr/local/lib/python3.6/dist-packages/flask/json/__init__.py", line 370, in jsonify
E                   dumps(data, indent=indent, separators=separators) + "\n",
E                 File "/usr/local/lib/python3.6/dist-packages/flask/json/__init__.py", line 211, in dumps
E                   rv = _json.dumps(obj, **kwargs)
E                 File "/usr/lib/python3.6/json/__init__.py", line 238, in dumps
E                   **kw).encode(obj)
E                 File "/usr/lib/python3.6/json/encoder.py", line 201, in encode
E                   chunks = list(chunks)
E                 File "/usr/lib/python3.6/json/encoder.py", line 430, in _iterencode
E                   yield from _iterencode_dict(o, _current_indent_level)
E                 File "/usr/lib/python3.6/json/encoder.py", line 404, in _iterencode_dict
E                   yield from chunks
E                 File "/usr/lib/python3.6/json/encoder.py", line 404, in _iterencode_dict
E                   yield from chunks
E                 File "/usr/lib/python3.6/json/encoder.py", line 437, in _iterencode
E                   o = _default(o)
E                 File "/usr/local/lib/python3.6/dist-packages/flask/json/__init__.py", line 100, in default
E                   return _json.JSONEncoder.default(self, o)
E                 File "/usr/lib/python3.6/json/encoder.py", line 180, in default
E                   o.__class__.__name__)
E               TypeError: Object of type 'set' is not JSON serializable
```

### LINKED ISSUE #160: Remove unnecessary LIFERAY_PORT from tests
E.g. https://github.com/jirik/layman/blob/e92333167df3d5e8b85879a125f2535b7275cb78/src/layman/layer/rest_layer_patch_test.py#L14

### PR DESCRIPTION
**Pull request by index-git**
Go per commit
Closes #160 
Closes #157

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,8 @@
 - [#28](https://github.com/jirik/layman/issues/28) New environment variable [LAYMAN_PRIME_SCHEMA](doc/env-settings.md#LAYMAN_PRIME_SCHEMA). 
 - [#28](https://github.com/jirik/layman/issues/28) New REST endpoint [GET Users](doc/rest.md#get-users) with list of all users registered in Layman.
 - [#28](https://github.com/jirik/layman/issues/28) New attribute `access_rights` added to [GET Layers](doc/rest.md#get-layers), [GET Layer](doc/rest.md#get-layer), [GET Maps](doc/rest.md#get-maps) and [GET Map](doc/rest.md#get-map) responses. New parameters `access_rights.read` and `access_rights.write` added to [POST Layers](doc/rest.md#post-layers), [PATCH Layer](doc/rest.md#patch-layer), [POST Maps](doc/rest.md#post-maps) and [PATCH Map](doc/rest.md#patch-map) requests.
+- [#28](https://github.com/jirik/layman/issues/28) At first start of Layman, access rights are set so [that only owner of the workspace can read or edit them](doc/security.md#Authorization) for publications in [private workspaces](doc/models.md#Workspace) and [that anyone can read or edit](doc/security.md#Authorization) for publications in [public workspaces](doc/models.md#Workspace).
```
Comment (jirik) [inline CHANGELOG.md:21]: "that only owner of the workspace can read or edit them" tohle myslím není chování, které chceme. U existujících vrstev v personal chceme zachovat REWO, protože dosud měly REWO.

V kódu je to myslím správně, špatně je to jen v dokumentaci, ale raději to prosím ověř.

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,11 +11,13 @@
    - `LAYMAN_AUTHZ_MODULE=layman.authz.read_everyone_write_owner` (variable to remove)
       - `GRANT_CREATE_PUBLIC_WORKSPACE=` (new variable)
       - `GRANT_PUBLISH_IN_PUBLIC_WORKSPACE=` (new variable)
+- [#101](https://github.com/jirik/layman/issues/101) Change [LAYMAN_CLIENT_VERSION](doc/env-settings.md#LAYMAN_CLIENT_VERSION) from `v1.3.0` to `v1.4.0`
```
Comment (jirik) [inline CHANGELOG.md:14]: Tuhle změnu nevidím v .env filech, ani nevidím daný release. Jeslti je to myšleno jen jako příprava, raději bych s ní počkal, až to bude implementováno.

**Review by jirik — CHANGES_REQUESTED** [review]
Commit 833051a44bfd6b4b1ef33300079d90239c32054b should reference issue https://github.com/jirik/layman/issues/157. Then this PR is ready.

**Comment by index-git** [comment]
> Commit [833051a](https://github.com/jirik/layman/commit/833051a44bfd6b4b1ef33300079d90239c32054b) should reference issue #157. Then this PR is ready.

Linked with both #157 and #160

**Comment by jirik** [comment]
> > Commit [833051a](https://github.com/jirik/layman/commit/833051a44bfd6b4b1ef33300079d90239c32054b) should reference issue #157. Then this PR is ready.
> 
> Linked with both #157 and #160

Good point with #160. If you plan to rebase (not squash) this PR, it;s better to add those references to particular commit messages, not just to PR.

**Comment by index-git** [comment]
> > > Commit [833051a](https://github.com/jirik/layman/commit/833051a44bfd6b4b1ef33300079d90239c32054b) should reference issue #157. Then this PR is ready.
> > 
> > 
> > Linked with both #157 and #160
> 
> Good point with #160. If you plan to rebase (not squash) this PR, it;s better to add those references to particular commit messages, not just to PR.

Done.

**Review by jirik — APPROVED** [review]
